internal/telemetry/tracker.go

```diff
@@ -0,0 +1,78 @@
+package telemetry
+
+import (
+	"fmt"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+// Service tracks events
+type Service interface {
+	TrackEvent(eventType EventType, data ...EventData)
+}
+
+type service struct {
+	userID      string
+	executionID string
+	command     string
+	tracker     tracker
+}
+
+// NewService creates a new service of type mode with userID and
+// executionID appended to each event it tracks
+func NewService(mode Mode, userID string, executionID string, command string) Service {
+	service := service{userID: userID, executionID: executionID, command: command}
+	switch mode {
+	case ModeOn, ModeNil:
+		service.tracker = &segmentTracker{}
+	case ModeStdout:
+		service.tracker = &stdoutTracker{}
+	case modeTest:
```

Comment:
likewise avoiding this `case` here, in the test we should be able to create a `&service` directly.  and the `testTracker` struct definition can live in the test file